### ISSUE TITLE
Danbooru doesn't work

### ISSUE THREAD

**Issue by Valkyncirael** [Bionus/imgbrd-grabber#2912]
Danbooru.donmai.us as a source doesn't work again. Any search done only brings in the result of "possible reasons: server offline".

Using your own user-agent in "Headers" no longer works either. Tried reinstalling a fresh v.7.10.1x64 just in case, both zip. and exe. but nothing. 

Just a red error code for Cloudflare wall in the log, followed by a yellow "Warning: no valid source of the site returned result". Same with Html, Json and Xml, all three basic options.

**Comment by GaryOni** [comment]
Same

**Comment by bgilc** [comment]
Just FYI:
https://danbooru.donmai.us/forum_topics/23284

"Bots or scripts that try to impersonate browsers are now blocked. I get too much traffic from people doing things like trying to scrape the entire site one post at a time, or checking the front page 10 times per second, or checking hundreds of tags every few minutes. It's hard to control this traffic because people do things like use tons of IPs to avoid rate limits, or rotate IPs to avoid bans, or spoof browsers to blend in with normal traffic. So now anything pretending to be a browser is blocked so I can better keep track of API traffic.

You can either set your user agent to something unique, that's not pretending to be a browser, or if you can't do that, ask the developer of the program to."

**Comment by GaryOni** [comment]
https://danbooru.donmai.us/forum_topics/22838

"Grabber works for me after creating a header titled "user-agent" and setting it's value as my username."

This one worked.

**Comment by lumiamilk** [comment]
> https://danbooru.donmai.us/forum_topics/22838
> 
> "Grabber works for me after creating a header titled "user-agent" and setting it's value as my username."
> 
> This one worked.

wow it works ! thank you.

**Comment by MalOMaw** [comment]
It does not work when you add api-key which is not satisfying at all

**Comment by LordOfLA** [comment]
yeah, I just set the user-agent to grabber/7.10.1 since my ruby scripts to sort/delete images I download worked just fine.

**Comment by JenXIII** [comment]
Workaround to API Key not working seems to be use your session cookie from your (logged in) browser
